### PR TITLE
*: provide a command line flag --skip-grant-table

### PR DESCRIPTION
**Pull request by tiancaiamao**
This flag is useful when user forget root password. 
A powerful and dangerous flag.

### CODE REVIEW

File: session_test.go [hunk 1]
```diff
@@ -2079,6 +2080,29 @@ func (s *testSessionSuite) TestSessionAuth(c *C) {
...
 }
 
+func (s *testSessionSuite) TestSkipWithGrant(c *C) {
+	defer testleak.AfterTest(c)()
+	dbName := "test_skip_with_grant"
+	se := newSession(c, s.store, dbName)
+	dropDBSQL := fmt.Sprintf("drop database %s;", dbName)
+
+	save1 := privileges.Enable
+	save2 := privileges.SkipWithGrant
```
Comment (hanfei1991) [inline session_test.go]: defer privileges.Enable = save1, privileges.Enable = save1

**Comment by shenli** [comment]
MySQL only allow root user to skip authentication.

File: session_test.go (hunk 1 above)
Comment (tiancaiamao) [inline session_test.go]: defer can't apply to a expression, it need to use a function.
here is a really simple case, defer seems overkill.

**Comment by tiancaiamao** [comment]
I checked that the user have root privilege. @shenli

**Comment by coocood** [comment]
LGTM

**Pull request merged**